sdlib/d/gc/arena.d

```diff
@@ -374,6 +448,39 @@ private:
 		unusedExtents.insert(e);
 	}
 
+	int resizePagesImpl(Extent* e, uint n, uint oldPages, uint newPages) {
+		assert(mutex.isHeld(), "Mutex not held!");
+		assert(oldPages > 0 && oldPages <= PageCount,
+		       "Invalid number of old pages!");
+		assert(newPages > 0 && newPages <= PageCount,
+		       "Invalid number of new pages!");
+		assert(n <= PageCount - max(oldPages, newPages), "Invalid index!");
+
+		if (oldPages == newPages) {
+			return 0;
+		}
```

Comment:
We shouldn't have reached this point if this was the case, right? Why take the mutex in such a trivial situation?